File: mcorg.backends/project-service/src/main/java/app/mcorg/project/presentation/rest/entities/project/CreateProjectRequest.java
package app.mcorg.project.presentation.rest.entities.project;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotEmpty;

public record CreateProjectRequest(
        @NotEmpty(message = "Project needs to be in a world")
        String worldId,
        @NotEmpty(message = "Project needs to be in a team")
        String teamId,
        @NotEmpty(message = "Project name cannot be empty")
        @Schema(example = "Project Name")
        String name) {
}

File: webapp/src/main/kotlin/app/mcorg/presentation/entities/project/CreateProjectRequest.kt
package app.mcorg.presentation.entities.project

import app.mcorg.domain.model.minecraft.Dimension
import app.mcorg.domain.model.projects.Priority

data class CreateProjectRequest(
    val name: String,
    val priority: Priority,
    val dimension: Dimension,
    val requiresPerimeter: Boolean
)